src/blas/impl/KokkosBlas2_serial_gemv_impl.hpp

```diff
@@ -0,0 +1,292 @@
+//
+//                        Kokkos v. 3.0
+//       Copyright (2020) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Siva Rajamanickam (srajama@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+#ifndef __KOKKOSBLAS_GEMV_SERIAL_IMPL_HPP__
+#define __KOKKOSBLAS_GEMV_SERIAL_IMPL_HPP__
+
+/// \author Kyungjoo Kim (kyukim@sandia.gov)
+
+#include "KokkosBlas_util.hpp"
+#include "KokkosBlas2_serial_gemv_internal.hpp"
+#include "KokkosBatched_Vector.hpp"  // for Vector handling in CompactMKL impls
+
+namespace KokkosBlas {
+
+template <typename ArgTrans, typename ArgAlgo>
+struct SerialGemv {
+  template <typename ScalarType, typename AViewType, typename xViewType,
+            typename yViewType>
+  KOKKOS_INLINE_FUNCTION static int invoke(const ScalarType /*alpha*/,
+                                           const AViewType & /*A*/,
+                                           const xViewType & /*x*/,
+                                           const ScalarType /*beta*/,
+                                           const yViewType & /*y*/) {
+    Kokkos::abort("Error: encounter dummy impl");
+    return 0;
+  }
+};
+
+///
+/// Serial Impl
+/// ===========
+/// CompactMKL does not exist on Gemv
```

Comment:
This needs to go in `src/impl/tpls` which is where TPLs are handled.

Reply:
Of course, I was happy to split out the MKL implementation into dedicated header on https://github.com/kokkos/kokkos-kernels/pull/1433/commits/8d80d3553c91060a7843ef805855ec53beedaf83.